File: app/src/main/java/com/example/miodragmilosevic/nativetest/filters/javaFilters/NoiseFilter.java
package com.example.miodragmilosevic.nativetest.filters.javaFilters;

import android.graphics.Color;

import com.example.miodragmilosevic.nativetest.filters.ImageFilter;

import java.util.Random;

/**
 * Created by miodrag.milosevic on 12/15/2017.
 */

public class NoiseFilter extends ImageFilter {
    private static final int COLOR_MAX = 255;

    @Override
    protected int[] transformPixels(int[] pixelArray, int imageWidth, int imageHeight) {
        Random random = new Random();

        // iteration through pixels
        for (int i = 0; i < pixelArray.length; i++) {
            int randColor = Color.rgb(random.nextInt(COLOR_MAX),
                    random.nextInt(COLOR_MAX), random.nextInt(COLOR_MAX));
            // OR
            pixelArray[i] |= randColor;
        }
        return pixelArray;
    }
}

File: app/src/main/cpp/NoiseFilter.cpp
//
// Created by miodrag.milosevic on 12/29/2017.
//

#include <stdlib.h>
#include "NoiseFilter.h"
#include "color.h"

void NoiseFilter::transformPixels(int *pixelArray, int imageWidth, int imageHeight) {

    // iteration through pixels
    for (jint i = 0; i < imageWidth*imageHeight; i++) {
        int randColor = Color::rgb(rand()%COLOR_MAX,
                                   rand()%COLOR_MAX, rand()%COLOR_MAX);
        // OR
        pixelArray[i] |= randColor;
    }
}
